tsi-typescript.el

```diff
@@ -321,19 +346,37 @@
      (or child-indentation 0)
      (or comment-indentation 0))))
 
-
 (defun tsi--current-line-empty-p ()
   (string-match-p "\\`[[:space:]]*$" (thing-at-point 'line)))
 
+(defun tsi-typescript--tsc-node-type-or-nil (maybe-node)
+  "Return node type if maybe-node is not nil, else nil.
+
+Avoids '(wrong-type-argument user-ptrp nil)' error if maybe-node is nil.
+"
+  (if maybe-node
+      (tsc-node-type maybe-node)
+    nil))
+
 (defun tsi-typescript--get-indent-for-current-line ()
   (let* ((node-at-point (tree-sitter-node-at-point))
-         (current-type (tsc-node-type node-at-point)))
+         (current-type (tsc-node-type node-at-point))
+         (parent-type (tsi-typescript--tsc-node-type-or-nil (tsc-get-parent node-at-point))))
```

Comment:
can we avoid the extra utility function if we use `when-let*` here instead of `let*`?  I assume that `(tsc-get-parent node-at-point)` only returns nil if point is on a `program` node, in which case indentation is guaranteed to be 0, right?